src/model.jl

```diff
@@ -335,27 +379,51 @@ function _add_constraints_demand_flexibility!(
 )
     if demand_flexibility.rolling_balance
         println("rolling load balance: ", Dates.now())
-        JuMP.@constraint(
-            m,
-            rolling_load_balance[
-                i in 1:sets.num_load_bus,
-                k in 1:(interval_length - demand_flexibility.duration)
-            ],
-            sum(
-                m[:load_shift_up][i, j] - m[:load_shift_dn][i, j]
-                for j in k:(k + demand_flexibility.duration)
-            ) >= 0,
-        )
+        if demand_flexibility.cost_input_type == "AREA"
+            JuMP.@constraint(
+                m,
+                rolling_load_balance[
+                    i in 1:sets.num_load_bus,
```

Comment:
For example, if we go with my suggestion from above, we could remove the `if`-`else` statement and just have this be `i in 1:sets.num_flexible_bus,`